src/main/java/seedu/address/model/student/YearMatchPredicate.java

```diff
@@ -0,0 +1,31 @@
+package seedu.address.model.student;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Student}'s {@code Year} matches the given year.
+ */
+public class YearMatchPredicate implements Predicate<Student> {
+
+    // Attributes
+    private final Year year;
+
+    // Constructor
+    public YearMatchPredicate(Year year) {
+        this.year = year;
+    }
+
+    @Override
+    public boolean test(Student student) {
+        return this.year.equals(student.getYear());
```

Comment:
`Year` is actually a String now that accepts alphanumeric characters. Hence I think we should relax the requirements to be one where it returns true if `student.getYear().year` contains the keyword (e.g. `year` could be a String and it is matched using `StringUtils#containsWordIgnoreCase`). 
That way, if you search "4" you could get those who put "Sec 4", "sec 4" or "Secondary 4" if the user is inconsistent.

It also means you will be less likely to land an `IllegalArgumentException` if the user parses an invalid string since right now `FindCommandParser` currently has to call `Year#new` to construct this.

Reply:
Thanks for mentioning this, the branch was created before the updates were pushed but I've now merged the changes from master! I've also updated the user guide to clarify the behaviour of this.